src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -107,6 +109,77 @@ public async Task<IActionResult> GetHostStatus([FromServices] IScriptHostManager
             return Ok(status);
         }
 
+        /// <summary>
+        ///  Currently, anyone in Reader role can access this information.
+        ///  If this API is extended to include any secrets, it will need to be
+        ///  locked down to only Contributor roles.
+        /// </summary>
+        [HttpGet]
+        [Route("admin/host/processes")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public async Task<IActionResult> GetWorkerProcesses([FromServices] IScriptHostManager scriptHostManager)
+        {
+            if (!Utility.TryGetHostService(scriptHostManager, out IWebHostRpcWorkerChannelManager webHostLanguageWorkerChannelManager))
+            {
+                return StatusCode(StatusCodes.Status503ServiceUnavailable);
+            }
+
+            var hostProcess = Process.GetCurrentProcess();
+            List<FunctionProcesses.FunctionProcessInfo> processes = new()
+            {
+                new FunctionProcesses.FunctionProcessInfo()
+                {
+                    ProcessId = hostProcess.Id,
+                    DebugEngine = RpcWorkerConstants.DotNetCoreDebugEngine,
+                    IsEligibleForOpenInBrowser = false,
+                    ProcessName = hostProcess.ProcessName
```

Comment:
Is the process name going to be dotnet.exe or the dll name? In that case, how does the caller know it is the host process or the oop worker process (in the case of dotnet isolated/powershell)? Is that a concern?

Reply:
This would likely be dotnet.exe. At the moment there is no differentiation between host and worker process. My assumption is that VS will provide the user with the prompt to select a process in case there is ambiguity. We can update the API incase a clear distinction is needed.